esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2946,55 +2850,43 @@ bool CFileSprayEx::onFileList(IEspContext &context, IEspFileListRequest &req, IE
                 throw MakeStringException(ECLWATCH_ACCESS_TO_FILE_DENIED, "Only cfg or log file allowed.");
         }
 
-        if (!validateDropZonePath(nullptr, netaddr, sPath) && !validateConfigurationDirectory(nullptr, "log", nullptr, nullptr, sPath)) //The path should be the absolute path for the dropzone or log file.
-            throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid file path %s", sPath.str());
-
-        RemoteFilename rfn;
-        SocketEndpoint ep;
-#ifdef MACHINE_IP
-        ep.set(MACHINE_IP);
-#else
-        ep.set(netaddr);
-        if (ep.isNull())
-            throw MakeStringException(ECLWATCH_INVALID_INPUT, "FileList: cannot resolve network IP from %s.", netaddr);
-#endif
-        rfn.setPath(ep, sPath.str());
-        Owned<IFile> f = createIFile(rfn);
-        if (f->isDirectory()!=fileBool::foundYes)
-            throw MakeStringException(ECLWATCH_INVALID_DIRECTORY, "%s is not a directory.", path);
-
-        IArrayOf<IEspPhysicalFileStruct> files;
-        Owned<IDirectoryIterator> di = f->directoryFiles(NULL, false, true);
-        if(di.get() != NULL)
+        IArrayOf<IConstPhysicalFileStruct>& files = resp.getFiles();
+        if (isEmptyString(dropZoneName) && validateConfigurationDirectory(nullptr, "log", nullptr, nullptr, sPath))
```

Comment:
> How is netaddr verified? Is there any authentication on who can view logs from where? e.g. could a rouge actor type in any IP and brows any /var/log/HPCCSystems logs from anywhere this way? How does this work in containerized? What is the route to getting similar logs in Containerized?

1. the netaddr will not be empty since the dropZoneName is empty (we check that before). 2. the existing function only do the feature authorization for listing logs. We do not authorize the access to different logs. It is the same as in WsTopology. 3. the feature we authorize here FILE_SPRAY_URL. We said we should move this function (list logs) to WsTopolgy and authorize that feature. 4. We do not provide a UI to view those logs in containerized. I guess that people should use some 3rd party tool to view log there.

Reply:
> check if netaddr suppplied
> throw an error that no dropzone named if not
> issue an operator error if netaddr and no dropzone name supplied in containerized
1. the netaddr will not be empty since the dropZoneName is empty. 2. I will LOG an MCoperatorWarning if netaddr and no dropzone name supplied in containerized.

Reply:
>4. We do not provide a UI to view those logs in containerized. I guess that people should use some 3rd party tool to view log there.

Sure, well we have the IRemoteLogAccess interface for that - which is the abstract way for dealing with centralized off-node storage, and the interface I believe that the ZAP report is already using in cloud(?).

Anyway, it means that a code path of this function ia redundant  in Containerized, but you wouldn't know that by looking at it..

As you said, this functionality should be split, but to make it clearer for now,
can you move the "validateConfigurationDirectory(nullptr, "log", nullptr, nullptr, sPath)" check to the front of the method,
and throw an error immediately if isContainerized(), as it is not supported.
Plus, add a comment re. the [current] dual purpose of this function etc.